homeassistant/components/generic_thermostat/climate.py

```diff
@@ -307,6 +307,11 @@ def _async_startup(*_):
         if not self._hvac_mode:
             self._hvac_mode = HVACMode.OFF
 
+    @property
+    def capability_attributes(self) -> dict[str, Any] | None:
+        """Return the capability attributes."""
+        return {CONF_PRESETS[k]: v for (k, v) in self._presets.items()}
```

Comment:
Is this what you mean by the [customize feature](https://www.home-assistant.io/docs/configuration/customizing-devices/)? That doesn't appear to allow for adding more than a predetermined set of variables, none of which would allow for storing this data.